GetWorksheets doesn't handle errors correctly

Hello again

The function `getWorksheets` doesn't handle errors correctly if I try to process a corrupted xlsx file.

I look into the code and I think `reject` should be called on stream.on('error') or zip callback with err.

```js
getWorksheets(options: IWorksheetOptions) {
    return new Promise<IWorksheet[]>((resolve, reject) => {
        function processWorkbook() {
            zip.stream('xl/workbook.xml', (err: any, stream: ReadStream) => {

               // Here
               if (err) {
                return reject(err);
               }

                stream.pipe(saxStream({
                    strict: true,
                    tag: ['x:sheet', 'sheet'],
                })).on('data', (x: any) => {
                    sheets.push({
                        name: x.record.attribs.name,
                        hidden: x.record.attribs.state && x.record.attribs.state === 'hidden' ? true : false,
                    });
                });
                // Here
                stream.on('error', reject);

                stream.on('end', () => {
                    zip.close();
                    resolve(sheets);
                });
            });
        }

        let sheets: IWorksheet[] = [];
        const zip = new StreamZip({
            file: options.filePath,
            storeEntries: true,
        });
        zip.on('ready', () => {
            processWorkbook();
        });
    });
}
```


My error stack: 

```
[14:40:07] [ERROR] Error [ERR_UNHANDLED_ERROR]: Unhandled error. ('Bad archive')
    at StreamZip.emit (events.js:304:17)
    at StreamZip.EventEmitter.emit (domain.js:482:12)
    at StreamZip.emit (/var/www/my-app/node_modules/node-stream-zip/node_stream_zip.js:598:33)
    at FsRead.readUntilFoundCallback [as callback] (/var/www/my-app/node_modules/node-stream-zip/node_stream_zip.js:201:25)
    at FsRead.readCallback (/var/www/my-app/node_modules/node-stream-zip/node_stream_zip.js:869:21)
    at FSReqCallback.wrapper [as oncomplete] (fs.js:516:5)

```

Edit: I tried to put a try/catch block on my call to getWorksheets but it doesn't work.

--- a/src/index.ts
+++ b/src/index.ts
@@ -377,6 +377,9 @@ export function getWorksheets(options: IWorksheetOptions) {
     return new Promise<IWorksheet[]>((resolve, reject) => {
         function processWorkbook() {
             zip.stream('xl/workbook.xml', (err: any, stream: ReadStream) => {
+                if (err) {
+                    reject(err);
+                }
                 stream.pipe(saxStream({
                     strict: true,
                     tag: ['x:sheet', 'sheet'],
@@ -390,6 +393,7 @@ export function getWorksheets(options: IWorksheetOptions) {
                     zip.close();
                     resolve(sheets);
                 });
+                stream.on('error', reject);
             });
         }
 
@@ -401,6 +405,7 @@ export function getWorksheets(options: IWorksheetOptions) {
         zip.on('ready', () => {
             processWorkbook();
         });
+        zip.on('error', reject);
     });
 }
 

--- a/tests/assets/corrupted-file.xlsx
+++ b/tests/assets/corrupted-file.xlsx
@@ -1,0 +1,3 @@
+invalid-content, corrupted-file, column 3
+0,1,3
+1,2,3

--- a/tests/xlsx-stream.spec.ts
+++ b/tests/xlsx-stream.spec.ts
@@ -95,6 +95,12 @@ it('gets worksheets with correct hidden info', async (done) => {
     done();
 });
 
+it('get worksheets should fail gracefully for a corrupted xlsx', () => {
+    return getWorksheets({
+        filePath: './tests/assets/corrupted-file.xlsx',
+    }).catch(e => expect(e).toMatch('Bad archive'));
+});
+
 it('gets worksheet by name, even if they are reordered', async (done) => {
     const data: any = [];
     const stream = await getXlsxStream({

